src/js/media-controller.js

```diff
@@ -675,39 +879,38 @@ class MediaController extends MediaContainer {
     super.mediaUnsetCallback(media);
 
     // Remove all state change propagators
-    Object.keys(this._mediaStatePropagators).forEach((key) => {
-      const events = key.split(',');
+    Object.keys(MediaUIStates).forEach((key) => {
+      const {
+        mediaEvents,
+        rootEvents,
+        trackListEvents
+      } = MediaUIStates[key];
+
       const handler = this._mediaStatePropagators[key];
 
-      events.forEach((event) => {
-        const target =
-          event == fullscreenApi.event ? this.getRootNode() : media;
-        target.removeEventListener(event, handler);
+      mediaEvents?.forEach((eventName)=>{
+        media.removeEventListener(eventName, handler);
       });
-    });
 
-    Object.entries(this._textTrackMediaStatePropagators).forEach(
-      ([eventsStr, handler]) => {
-        const events = eventsStr.split(',');
-        events.forEach((event) => {
-          if (media.textTracks) {
-            media.textTracks.removeEventListener(event, handler);
-          }
-        });
-        handler();
-      }
-    );
+      rootEvents?.forEach((eventName)=>{
+        this.getRootNode().removeEventListener(eventName, handler);
+      });
+
+      trackListEvents?.forEach((eventName)=>{
+        media.textTracks?.removeEventListener(eventName, handler);
+      });
+    });
 
     // Reset to paused state
+    // TODO: Can we just reset all state here?
+    // Should hasPlayed refer to the media element or the controller?
+    // i.e. the poster might re-show if not handled by the poster el
     this.propagateMediaState(MediaUIAttributes.MEDIA_PAUSED, true);
   }
 
   propagateMediaState(stateName, state) {
-    if (arguments.length === 1) {
-      state = Delegates[stateName](this);
-    }
-
     propagateMediaState(this.mediaStateReceivers, stateName, state);
+    // TODO: I don't think we want these events to bubble? Video element states don't. (heff)
```

Comment:
@cjpillsbury confirming it's ok to turn off bubbling here. I don't have a strong defense but it feels wrong to bubble these state change events